records.py

```diff
@@ -282,40 +296,42 @@ def get_table_names(self, internal=False):
         # Setup SQLAlchemy for Database inspection.
         return inspect(self._engine).get_table_names()
 
-    def get_connection(self):
+    def get_connection(self, close_with_result=False):
         """Get a connection to this Database. Connections are retrieved from a
-        pool.
+        pool. By default, the retrieved connection remains open. Setting
+        close_with_result to True returns the connection to the pool once a
+        result is consumed.
         """
         if not self.open:
             raise exc.ResourceClosedError('Database closed.')
 
-        return Connection(self._engine.connect())
+        return Connection(self._engine.contextual_connect(close_with_result))
 
     def query(self, query, fetchall=False, **params):
         """Executes the given SQL query against the Database. Parameters can,
         optionally, be provided. Returns a RecordCollection, which can be
         iterated over to get result rows as dictionaries.
         """
-        with self.get_connection() as conn:
-            return conn.query(query, fetchall, **params)
+        conn = self.get_connection(True);
+        return conn.query(query, fetchall, **params)
 
     def bulk_query(self, query, *multiparams):
         """Bulk insert or update."""
 
-        with self.get_connection() as conn:
```

Comment:
Since Connections are not used as contextmanagers anymore, it is possible to remove the `__enter__`  and `__exit__` methods from `class Connection` and the tests still pass. 

I would vote for removing untested features.

Reply:
I decided to leave this in, as I think it may be a convenient feature even though it is not used internally. I added a test to verify the expected `Connection` context manager behavior.